### ISSUE TITLE
hide ServiceBinding from `odo service list`

### ISSUE THREAD

**Issue by kadel** [openshift/odo#4861]
## User Story
- As a developer
- I don't want to see ServiceBinding resource in `odo service list`
- Because technically it is not a service that my application can consume

## Acceptance Criteria
- `odo service list` should filter our ServiceBinding resources from the output.

/kind user-story
/priority medium
/area service
/area service-binding

**Comment by feloy** [comment]
Services are collected from boath places:
- devfile
- cluster

Services with a kind `ServiceBinding` should be ignored from both places

**Comment by dharmit** [comment]
@feloy do you think we will need to add some tests as a part of this PR? I can think of integration tests which ensure that we don't show `ServiceBinding/<name>` in the `odo service list` output.

**Comment by feloy** [comment]
> @feloy do you think we will need to add some tests as a part of the PR for this issue? I can think of integration tests which ensure that we don't show `ServiceBinding/<name>` in the `odo service list` output.

Yes, I can think of integration tests to verify that links are not visible, either when they are pushed and unpushed.

**Comment by dharmit** [comment]
> As the list of services is used internally to check which services should be deleted when running `odo push`, we have to take care to adapt this part of the code also.

Some work for this has already been started in #4905 where we first push the Operator backed services, then the Deployment and finally the link. Worth taking a look at it's code for this issue.

**Comment by mohammedzee1000** [comment]
Looking through PR linked by dharmit

**Comment by mohammedzee1000** [comment]
I will have the PR open first thing once the release work is done
